zephyr/CMakeLists.txt

```diff
@@ -1141,17 +1141,21 @@ elseif(CONFIG_DTS_CODEC)
 	endif()
 endif()
 
-if(CONFIG_WAVES_CODEC STREQUAL "m")
-	add_subdirectory(${SOF_AUDIO_PATH}/module_adapter/module/waves/llext
-			 ${PROJECT_BINARY_DIR}/waves_llext)
-	add_dependencies(app waves)
-elseif(CONFIG_WAVES_CODEC)
-	zephyr_library_sources(
-		${SOF_AUDIO_PATH}/module_adapter/module/waves/waves.c
-	)
-	zephyr_library_sources_ifdef(CONFIG_WAVES_CODEC_STUB
-		${SOF_AUDIO_PATH}/module_adapter/module/waves/maxx_stub.c
-	)
+if(CONFIG_WAVES_CODEC)
+	set(WAVES_DIR ${SOF_AUDIO_PATH}/module_adapter/module/waves)
+	if(CONFIG_WAVES_CODEC STREQUAL "m")
+		add_subdirectory(${WAVES_DIR}/llext ${PROJECT_BINARY_DIR}/waves_llext)
+		add_dependencies(app waves)
+	else()
+		zephyr_library_sources(${WAVES_DIR}/waves.c)
+		if(CONFIG_WAVES_CODEC_STUB)
+			zephyr_library_sources_ifdef(CONFIG_WAVES_CODEC_STUB
+						     ${WAVES_DIR}/maxx_stub.c)
+		else()
+			zephyr_library_import(waves_codec
+				${sof_top_dir}/third_party/lib/libMaxxChrome.a)
```

Comment:
do you think it would also be good to enable modular build with the library? https://github.com/thesofproject/sof/blob/41430f748433bc74c83e1e84b8e431f0cd1e5c3c/src/audio/module_adapter/module/waves/llext/CMakeLists.txt#L10

Reply:
Undeniably, I just don't know how the llext stuff works yet and was afraid to break anything.  :)

Reply:
That can be done as stage 2 - no reason to block this PR today.